src/detector/implementations/picamera2_implementation.py

```diff
@@ -0,0 +1,181 @@
+from src.detector.api import \
+    GetCaptureDeviceResponse, \
+    GetCaptureImageRequest, \
+    GetCaptureImageResponse, \
+    GetCapturePropertiesResponse, \
+    SetCaptureDeviceRequest, \
+    SetCapturePropertiesRequest
+from src.common import \
+    EmptyResponse, \
+    ErrorResponse, \
+    get_kwarg, \
+    MCastResponse
+from src.common.structures.capture_status import CaptureStatus
+
+from src.detector.implementations import AbstractCameraInterface
+
+from picamera2 import Picamera2, Preview
+
+import base64
+import cv2
+import datetime
+import logging
+import time
+import numpy
+import os
+from typing import Any, Callable
+
+logger = logging.getLogger(__name__)
+
+class PiCamera(AbstractCameraInterface):
+
+    _capture: numpy.ndarray | None
+    _captured_image: numpy.ndarray | None
```

Comment:
What is the difference between these?

Can you please add type hints for other class members?

Reply:
For the original code for the USB webcam there was a distinction between _capture and _captured_image. I copied that structure for the Pi camera implementation but can change it as it is now unnecessary.

I have type hints for capture_status and captured_timestamp_utc in the parent abstract class. Should I move/copy them to the child class?